### ISSUE TITLE
[Android] Emitted push event are not being handled on app launch

### ISSUE THREAD

**Issue by AntiHate** [urbanairship/react-native-module#148]
We are facing exactly the same issue in v2.0.1 as described in #58 but with push notification opened. We are using react-navigation with redux so its takes some time to initialize the listener.

We fixed this by making some changes in EventEmitter.java. 

```
ReactContext reactContext = reactInstanceManager.getCurrentReactContext();
        if (reactContext != null && reactContext.hasActiveCatalystInstance()) {
//HERE --> This execute when app launched after soft exit (hardware back button exit)
            synchronized (knownListeners) {
                if (knownListeners.contains(event.getName())) {
                    emit(reactContext, event.getName(), event.getBody());
                } else if(event.getName() == "com.urbanairship.notification_response") {
                    pendingEvents.add(event);
                }
            }
        } else if (reactInstanceManager.hasStartedCreatingInitialContext()) {
            reactInstanceManager.addReactInstanceEventListener(new ReactInstanceManager.ReactInstanceEventListener() {
                public void onReactContextInitialized(ReactContext reactContext) {
                    emit(reactContext, event.getName(), event.getBody());
                    reactInstanceManager.removeReactInstanceEventListener(this);
                }
            });
        } else {
//HERE --> This executes when app launched after killed (cold launch)
            if(event.getName() == "com.urbanairship.notification_response") {
                pendingEvents.add(event);
            }            
        }
```

I don't know much about native side but this block doesn't execute when app is opened after exit (soft or killed).

```
        // Force the call to be on the main thread
        if (Looper.getMainLooper() != Looper.myLooper()) {
            final Context applicationContext = context.getApplicationContext();
            new Handler(Looper.getMainLooper()).post(new Runnable() {
                @Override
                public void run() {
                    synchronized (knownListeners) {
                        if (knownListeners.contains(event.getName())) {
                            sendEvent(applicationContext, event);
                        } else {
                            pendingEvents.add(event);
                        }
                    }
                }
            });

            return;
        }
```

Any suggestions?

**Comment by rlepinski** [comment]
I think I see the problem. It looks like we only add to pending if sendEvents is called off the main thread. Ill have a PR in a bit to fix it.

**Comment by rlepinski** [comment]
@AntiHate I am working on the branch https://github.com/urbanairship/react-native-module/tree/GH-148, I just need to test it then I will post a PR.

**Comment by AntiHate** [comment]
Thanks @rlepinski for acknowledging the issue.

We just tested the GH-148 branch and its still not working for us, infact we're not receiving events for both notificationResponse and notificationResponse, not even when listeners are initialized and app is in foreground.

Let me know if you need me to try anything else. Thanks!

**Comment by rlepinski** [comment]
I probably made a bad typo somewhere. Testing now.

**Comment by ralepinski** [comment]
@AntiHate Found my typos. Could you give it another try? Ill post the PR now.

**Comment by AntiHate** [comment]
Sure, meanwhile let me see how to use PR in package.json. =/

**Comment by ralepinski** [comment]
By PR I meant Pull Request - https://github.com/urbanairship/react-native-module/pull/150

**Comment by AntiHate** [comment]
Yes but I'm not sure what link to use in package.json to use the PR.

I tried `"urbanairship-react-native": "github:urbanairship/react-native-module#pull/150"` but got install error.

**Comment by AntiHate** [comment]
it worked with `"urbanairship-react-native": "github:urbanairship/react-native-module#pull/150/head"`

**Comment by AntiHate** [comment]
Great! Every scenario seems to be working. Cold launch, launch after soft exit, background. Thanks @rlepinski for your help, really appreciate!

**Comment by rlepinski** [comment]
thanks for testing! hopefully we will get a release out today

**Comment by rlepinski** [comment]
Fixed in 2.0.2
